### PR TITLE
[BugFix](BE) fix condition index doesn't match

### LINKED ISSUE #11472: [Bug] select with condition will coredump
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

In thie case, BE will core when find condition by cid. Now, every rowset will store their segment, but in unique table, that is a hide column 'delete_sign' always at schema end. So after do schema change, this column index in each segment is different. 

```
DROP TABLE IF EXISTS schema_change_update_regression_test;

CREATE TABLE schema_change_update_regression_test (
                `user_id` LARGEINT NOT NULL COMMENT "用户id",
                `date` DATE NOT NULL COMMENT "数据灌入日期时间",
                `city` VARCHAR(20) COMMENT "用户所在城市",
                `age` SMALLINT COMMENT "用户年龄",
                `sex` TINYINT COMMENT "用户性别",
                `last_visit_date` DATETIME DEFAULT "1970-01-01 00:00:00" COMMENT "用户最后一次访问时间",
                `last_update_date` DATETIME DEFAULT "1970-01-01 00:00:00" COMMENT "用户最后一次更新时间",
                `last_visit_date_not_null` DATETIME NOT NULL DEFAULT "1970-01-01 00:00:00" COMMENT "用户最后一次访问时间",
                `cost` BIGINT DEFAULT "0" COMMENT "用户总消费",
                `max_dwell_time` INT DEFAULT "0" COMMENT "用户最大停留时间",
                `min_dwell_time` INT DEFAULT "99999" COMMENT "用户最小停留时间")
            UNIQUE KEY(`user_id`, `date`, `city`, `age`, `sex`) DISTRIBUTED BY HASH(`user_id`)
            PROPERTIES ( "replication_num" = "1" );

INSERT INTO schema_change_update_regression_test VALUES
             (1, '2017-10-01', 'Beijing', 10, 1, '2020-01-01', '2020-01-01', '2020-01-01', 1, 30, 20);

INSERT INTO schema_change_update_regression_test VALUES
             (2, '2017-10-01', 'Beijing', 10, 1, '2020-01-02', '2020-01-02', '2020-01-02', 1, 31, 21);

SELECT * FROM schema_change_update_regression_test order by user_id ASC, last_visit_date;

ALTER table schema_change_update_regression_test ADD COLUMN new_column INT default "1";

SELECT * FROM schema_change_update_regression_test order by user_id DESC, last_visit_date;
```

### What You Expected?

work

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Lchangliang**
# Proposed changes

Issue Number: close #11472

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/rowset/segment_v2/segment_iterator.h [hunk 1]
```diff
@@ -165,9 +165,9 @@ class SegmentIterator : public RowwiseIterator {
     const Schema& _schema;
     // _column_iterators.size() == _schema.num_columns()
     // _column_iterators[cid] == nullptr if cid is not in _schema
-    std::vector<ColumnIterator*> _column_iterators;
+    std::map<int32_t, ColumnIterator*> _column_iterators;
     // FIXME prefer vector<unique_ptr<BitmapIndexIterator>>
```
Comment (yiguolei) [inline be/src/olap/rowset/segment_v2/segment_iterator.h:169]: modify the comments and explain the map structure.

**Comment by morningman** [comment]
Please fill the Checklist

**Review by yiguolei — DISMISSED** [review]
LGTM

**Review by HappenLee — DISMISSED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**